packages/driver/cypress/integration/e2e/multi-domain/multi_domain_spec.ts

```diff
@@ -278,4 +287,223 @@ describe('multi-domain', { experimentalSessionSupport: true, experimentalMultiDo
 
     it('short circuits the secondary domain command queue when "done()" is called early')
   })
+
+  describe('yields', () => {
+    it('yields a value', () => {
+      cy.switchToDomain('foobar.com', () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('yields the cy value even if a return is present', () => {
+      cy.switchToDomain('foobar.com', async () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+
+        const p = new Promise((resolve, reject) => {
+          setTimeout(() => {
+            resolve('text')
+          }, 1000)
+        })
+
+        return p
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('errors if a cy command is present and it returns a sync value', (done) => {
+      cy.on('fail', (err) => {
+        assertLogLength(logs, 6)
+        expect(logs[5].get('error')).to.eq(err)
+        expect(err.message).to.include('`cy.switchToDomain()` failed because you are mixing up async and sync code.')
+
+        done()
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+
+        return 'text'
+      })
+    })
+
+    it('yields synchronously', () => {
+      cy.switchToDomain('foobar.com', () => {
+        return 'From a secondary domain'
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('yields asynchronously', () => {
+      cy.switchToDomain('foobar.com', async () => {
+        return new Promise((resolve: (val: string) => any, reject) => {
+          setTimeout(() => {
+            resolve('From a secondary domain')
+          }, 1000)
+        })
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('succeeds if subject cannot be serialized and is not accessed synchronously', () => {
+      cy.switchToDomain('foobar.com', () => {
+        return {
+          symbol: Symbol(''),
+        }
+      }).then((obj) => {
+        return 'object not accessed'
+      }).should('equal', 'object not accessed')
+    })
+
+    it('throws if subject cannot be serialized and is accessed synchronously', (done) => {
+      cy.on('fail', (err) => {
+        assertLogLength(logs, 6)
+        expect(logs[5].get('error')).to.eq(err)
+        expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of it\'s properties not being supported by the structured clone algorithm.')
+
+        done()
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        return {
+          symbol: Symbol(''),
+        }
+      }).should('equal', '')
```

Comment:
I assume the `should` is what's failing here but I think it might be more readable if it's more explicit in a `.then`

```suggestion
      }).then((obj) => {
        // This will fail access the symbol
        return obj.symbol
      }
```

Reply:
updated